File: benchmark.py
# Arithmetic operations
arith_sum = 0
for i in range(100000000):
    arith_sum += i * 2

# String concatenation
str_concat = ''
for i in range(100000):
    str_concat += 'test'

# Array operations
arr = []
for i in range(100000):
    arr.append(i)
arr_sum = sum(arr)

# Function calls
def square(n):
    return n * n

func_sum = 0
for i in range(1000000):
    func_sum += square(i)

# File write and read 1000 times
for _ in range(1000):
    with open('test.txt', 'w') as f:
        f.write('This is a test file.\n' * 10000)
    with open('test.txt', 'r') as f:
        file_content = f.read()

# Sorting
import random
unsorted_arr = [random.random() for _ in range(100000)]
sorted_arr = sorted(unsorted_arr)

# Dictionary operations
dict = {f'key{i}': i for i in range(100000)}
dict_sum = sum(dict.values())

File: benchmark.js
import fs from 'fs';

// Arithmetic operations
let arithSum = 0;
for (let i = 0; i < 100000000; i++) {
    arithSum += i * 2;
}

// String concatenation
let strConcat = '';
for (let i = 0; i < 100000; i++) {
    strConcat += "test";
}

// Array operations
let arr = [];
for (let i = 0; i < 100000; i++) {
    arr.push(i);
}
let arrSum = arr.reduce((acc, val) => acc + val, 0);

// Function calls
function square(n) {
    return n * n;
}
let funcSum = 0;
for (let i = 0; i < 1000000; i++) {
    funcSum += square(i);
}

// File write and read 1000 times
for (let j = 0; j < 1000; j++) {
    fs.writeFileSync('test.txt', 'This is a test file.\n'.repeat(10000));
    let fileContent = fs.readFileSync('test.txt', 'utf8');
}

// Sorting
let unsortedArr = [];
for (let i = 0; i < 100000; i++) {
    unsortedArr.push(Math.random());
}
let sortedArr = unsortedArr.sort();

// Dictionary (object) operations
let dict = {};
for (let i = 0; i < 100000; i++) {
    dict[`key${i}`] = i;
}
let dictSum = 0;
for (let key in dict) {
    dictSum += dict[key];
}
